Folder restructure

chikeichan opened the pull request:
<img width="372" alt="Screen Shot 2019-03-14 at 1 50 32 AM" src="https://user-images.githubusercontent.com/8507735/54343233-9a56d300-45fb-11e9-9a90-453d54a83150.png">

❤️

chikeichan committed:
Remove ui/app/keychains/

chikeichan committed:
Remove ui/app/img/ (unused images)

chikeichan committed:
Move conversion-util to helpers/utils/

chikeichan committed:
Move token-util to helpers/utils/

chikeichan committed:
Move /helpers/*.js inside /helpers/utils/

chikeichan committed:
Move util tests inside /helpers/utils/

chikeichan committed:
Renameand move confirm-transaction/util.js to helpers/utils/

chikeichan committed:
Move higher-order-components to helpers/higher-order-components/

chikeichan committed:
Move infura-conversion.json to helpers/constants/

chikeichan committed:
Move all utility functions to helpers/utils/

chikeichan committed:
Move pages directory to top-level

chikeichan committed:
Move all constants to helpers/constants/

chikeichan committed:
Move metametrics inside helpers/

chikeichan committed:
Move app and root inside pages/

chikeichan committed:
Move routes inside helpers/

chikeichan committed:
Re-organize ducks/

chikeichan committed:
Move reducers to ducks/

chikeichan committed:
Move selectors inside selectors/

chikeichan committed:
Move test out of test folder

chikeichan committed:
Move action, reducer, store inside store/

chikeichan committed:
Move ui components inside ui/

chikeichan committed:
Move UI components inside ui/

chikeichan committed:
Move connected components inside components/app/

chikeichan committed:
Move i18n-helper inside helpers/

chikeichan committed:
Fix unit tests

chikeichan committed:
Fix unit test

whymarrh commented:
I know this I should've raised this earlier, but after looking at the image in the OP, can we drop `ui/app/`? Directories like `ui/app/components/app/` and `ui/app/ducks/app/` seem a bit redundant. We could instead have `ui/components/` and `ui/ducks/`. Thoughts?

danjm reviewed (changes requested):
QA'd and everything seems good.

Inspected the folder structure via my text editor. Its definitely a big improvement.

One preference I have that differs from what is there:

I like having `app.js`, `root.js`, `reducers.js` and `index.js` in the same place. These pull together all the high level pieces of the app and are a good starting point to understanding the app as a whole. So I think there are benefits for code readability if they are are all together.

Independent of that concern, I would add that `app.js` and `root.js` don't really fit in `pages/`. That directory makes sense as a "directory of all components that have their own route". Whereas `app.js` is more of a routing component and `root.js` is boilerplate react needed for the app to work. Perhaps put these one directory up: `app/app.js` and `app/root.js` `app.js` may need a renaming if this is done.

Other than these relatively minor concerns, I think this is good to merge.

chikeichan committed:
Move pages components

chikeichan committed:
Rename routes component

chikeichan committed:
Move reducers to ducks/index

chikeichan commented:
@danjm great suggestions - based on them:
- changed `pages/app.js` to `pages/routes/index.js`
- changed `pages/root.js` to `pages/index.js`
- move all `pages/*.js` to `pages/[component-name]/index.js`
- move `store/reducers.js` to `ducks/index.js`

I `actions.js` in the `store/` directory. This is the file to be re-organize into appropriate ducks.

chikeichan committed:
Fix bad path in unit test
